### PR TITLE
BF: Reject invalid project URLs

### LINKED ISSUE #139: git-remote helper ignores path in URL
Originally reported here: https://github.com/datalad/datalad/issues/5387

Culprit:
Datalad tries to clone a subdataset by appending the path to the URL of the subdataset. This would be fine if it simply fails and `clone` proceeds with the next candidate URL. However, it isn't properly failing. Instead git-remote helper only uses `urlparse(url).netloc` to instantiate the project object, ignoring there's `urlparse(url).path`, too, which if it was appended would lead to correct failure. This leads to `osf://ID/whatever` being treated as `osf://ID/`. In case of such a subdataset (as shown in the original issue) we clone the superdataset instead.

As far as I can tell, there's no case in which a URL like `https://osf.io/jr5vx/some/path` would ever lead to a project. So, I'm going to push a fix where git-remote will fail early on such a URL.
If it turns out, that such a URL could indeed be valid, then we would need to pass `netloc + path` to osf client and let it fail whenever it wants to.

### PR DESCRIPTION
**Pull request by bpoldrack**
Closes #139

### CODE REVIEW

**Comment by bpoldrack** [comment]
I assume the github action for windows can be disabled with windows being tested on appveyor or is there a reason for it still lingering around, @mih?

**Comment by mih** [comment]
> I assume the github action for windows can be disabled with windows being tested on appveyor or is there a reason for it still lingering around, @mih?

It can go AFAICS

File: datalad_osf/git_remote.py [hunk 1]
```diff
@@ -55,6 +55,8 @@ def __init__(self,
           Stream for logging.
         """
         self.parsed_url = urlparse(url)
+        if self.parsed_url.path:  # project urls have no path
+            raise RuntimeError("Invalid OSF project URL: %s" % url)
```
Comment (mih) [inline datalad_osf/git_remote.py:59]: This message is not very useful. It should convey what users need to do.

**Review by mih — CHANGES_REQUESTED** [review]

File: datalad_osf/git_remote.py (hunk 1 above)
Comment (bpoldrack) [reply datalad_osf/git_remote.py]: Done. Also check PR #143.

**Pull request merged**